docs/sources/flow/reference/cli/convert.md

```diff
@@ -0,0 +1,33 @@
+---
+title: grafana-agent convert
+weight: 100
+---
+
+# `grafana-agent convert` command
+
+The `grafana-agent convert` command converts a supported configuration file format
+to a Grafana Agent Flow river file.
+
+## Usage
+
+Usage: `grafana-agent convert [FLAG ...] FILE_NAME`
+
+If the `FILE_NAME` argument is not provided or if the `FILE_NAME` argument is
+equal to `-`, `grafana-agent convert` converts the contents of standard input. Otherwise,
+`grafana-agent convert` reads and converts the file from disk specified by the argument.
+
+The `--write` flag can be specified to write the contents of the converted
+file on disk with the converted results. `--write` can only be provided when
+`grafana-agent convert` is not reading from standard input.
+
+The command fails if the file being converted has syntactically incorrect
+configuration or cannot be converted to a Grafana Agent Flow river file.
+
+The following flags are supported:
+
+* `--write`, `-w`: Write the converted file back to disk when not reading from
+  standard input.
```

Comment:
Have we considered an `--output` flag instead? (and writing to stdout when it's not present)

IMHO overwriting the input configuration file might be confusing to users, cannot be retried, plus we have to keep track where the file came to check whether it can be used or not.

Reply:
I agree, updated